app/Models/Beatmap.php

```diff
@@ -251,6 +251,11 @@ public function canBeConvertedTo(int $rulesetId)
         return $this->playmode === static::MODES['osu'] || $this->playmode === $rulesetId;
     }
 
+    public function expireTopTagIds()
+    {
+        cache_expire_with_fallback("beatmap_top_tag_ids:{$this->getKey()}", $GLOBALS['cfg']['osu']['tags']['beatmap_tags_cache_duration']);
```

Comment:
is there reason why the tags are cached with fallback (which means it may potentially return empty array)? while the query isn't simple, I doubt it's slow enough to warrant the downside.